### PR TITLE
6788481: CellEditorListener.editingCanceled is never called

### PR DESCRIPTION
**Pull request by prsadhuk**
When editing of a table cell is canceled, the function editingCanceled of the registered listener CellEditorListener is not called as actionPerformed on ESC key press was not notifying the "cancel" listeners.
Fix is to handle "Cancel" action in actionPerformed() by forwarding the Cancel message from CellEditor to the delegate so that it can call `AbstractCellEditor.fireEditingCanceled(`) which notifies all listeners of cancel event.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-6788481](https://bugs.openjdk.org/browse/JDK-6788481): CellEditorListener.editingCanceled is never called

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to [02c8fc00](https://git.openjdk.org/jdk/pull/10964/files/02c8fc00eeb6e736d0aeac17884622630b2aeaaf)
 * [Alexander Zvegintsev](https://openjdk.org/census#azvegint) (@azvegint - **Reviewer**) ⚠️ Review applies to [02c8fc00](https://git.openjdk.org/jdk/pull/10964/files/02c8fc00eeb6e736d0aeac17884622630b2aeaaf)
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**) ⚠️ Review applies to [3efbfa7b](https://git.openjdk.org/jdk/pull/10964/files/3efbfa7bebf5aada079023eb6f2b7730de9defa4)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10964/head:pull/10964` \
`$ git checkout pull/10964`

Update a local copy of the PR: \
`$ git checkout pull/10964` \
`$ git pull https://git.openjdk.org/jdk pull/10964/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10964`

View PR using the GUI difftool: \
`$ git pr show -t 10964`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10964.diff">https://git.openjdk.org/jdk/pull/10964.diff</a>

</details>

### CODE REVIEW

**Review by prrace — APPROVED** [review]

**Comment by prrace** [comment]
I assume you ran all JTable tests (at leas) not just this new test.

**Comment by prsadhuk** [comment]
> I assume you ran all JTable tests (at leas) not just this new test.

I normally run the full clientlibs CI job which was ok.

**Review by azvegint — APPROVED** [review]

File: test/jdk/javax/swing/JTable/BugCellEditorListener.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import javax.swing.event.ChangeEvent;
+
+public class BugCellEditorListener {
+    static Robot robot;
+    static JFrame frame;
+    static JTable table;
+    static volatile Point pt;
+    static volatile boolean cancelled;
+
+    public static void main(String [] args) throws Exception {
```
Comment (mrserb) [inline test/jdk/javax/swing/JTable/BugCellEditorListener.java:51]: Can the test validates all installed L&F and not only default metal/aqua?

Comment (prsadhuk) [reply test/jdk/javax/swing/JTable/BugCellEditorListener.java:64]: test updated

**Review by mrserb — APPROVED** [review]

**Pull request closed without merging**